File: Data Structures & Algorithms/binary-search/submission-2.java
class Solution {

    private int binSearch(int[] nums,int beg,int end,int target)
    {
        if(beg > end)
            return -1;
        
        int mid = beg + (end - beg)/2;
        if(nums[mid] == target)
            return mid;
        else if(nums[mid] > target)
            return binSearch(nums,beg,mid - 1,target);
        else
            return binSearch(nums,mid + 1,end,target);
    }
    public int search(int[] nums, int target) {
        
        return binSearch(nums,0,nums.length-1,target);
    }
}

File: Data Structures & Algorithms/two-integer-sum-ii/submission-2.py
class Solution:
    def twoSum(self, num: List[int], target: int) -> List[int]:
        
        beg,end = 0,len(num) - 1
        while beg < end:
            sum = num[beg] + num[end]
            if sum == target:
                return [beg + 1, end + 1]
            elif sum > target:
                end -= 1
            else:
                beg += 1
